book/visual-effects.md

```diff
@@ -508,21 +508,17 @@ rectangle (e.g., `left`) were checked, replace them with the
 corresponding function on a Skia `Rect` (e.g., `left()`). Also replace
 calls to `containsPoint` with Skia's `contains`.
 
-While we're here, let's also add a `rect` field to the other drawing
-commands, replacing its `top`, `left`, `bottom`, and `right`
-fields:
-
 ``` {.python}
 class DrawText:
     def __init__(self, x1, y1, text, font, color):
+        self.rect = skia.Rect.MakeLTRB(
```

Comment:
This doesn't match the source code. You can run `make test` from the commandline to see the differences and fix them.